components/omnibox/browser/brave_history_url_provider_unittest.cc

```diff
@@ -0,0 +1,146 @@
+#include "base/test/bind.h"
+#include "base/test/task_environment.h"
+#include "base/time/time.h"
+#include "brave/components/omnibox/browser/brave_history_quick_provider.h"
+#include "brave/components/omnibox/browser/brave_omnibox_prefs.h"
+#include "build/build_config.h"
+#include "components/bookmarks/test/test_bookmark_client.h"
+#include "components/history/core/browser/history_service.h"
+#include "components/history/core/browser/url_database.h"
+#include "components/history/core/test/history_service_test_util.h"
+#include "components/omnibox/browser/autocomplete_match.h"
+#include "components/omnibox/browser/autocomplete_provider.h"
+#include "components/omnibox/browser/autocomplete_provider_listener.h"
+#include "components/omnibox/browser/autocomplete_result.h"
+#include "components/omnibox/browser/fake_autocomplete_provider.h"
+#include "components/omnibox/browser/fake_autocomplete_provider_client.h"
+#include "components/omnibox/browser/in_memory_url_index.h"
+#include "components/omnibox/browser/test_scheme_classifier.h"
+#include "components/omnibox/browser/verbatim_match.h"
+#include "components/prefs/pref_registry_simple.h"
+#include "components/prefs/pref_service.h"
+#include "components/prefs/testing_pref_service.h"
+#include "components/search_engines/default_search_manager.h"
+#include "components/search_engines/omnibox_focus_type.h"
+#include "components/search_engines/template_url.h"
+#include "components/search_engines/template_url_service.h"
+#include "components/url_formatter/url_fixer.h"
+#include "testing/gtest/include/gtest/gtest.h"
+#include "third_party/metrics_proto/omnibox_event.pb.h"
+#include "third_party/metrics_proto/omnibox_input_type.pb.h"
+#include "url/gurl.h"
+
+class BraveHistoryURLProviderTest : public testing::Test,
+                                    public AutocompleteProviderListener {
+ public:
+  BraveHistoryURLProviderTest() { HistoryQuickProvider::set_disabled(true); }
+
+  ~BraveHistoryURLProviderTest() override {
+    HistoryQuickProvider::set_disabled(false);
+  }
+
+  BraveHistoryURLProviderTest(const BraveHistoryURLProviderTest&) = delete;
+  BraveHistoryURLProviderTest& operator=(const BraveHistoryURLProviderTest&) =
+      delete;
+
+  AutocompleteInput CreateAutocompleteInput(base::StringPiece text) {
+    AutocompleteInput input(base::UTF8ToUTF16(text),
+                            metrics::OmniboxEventProto::OTHER,
+                            TestSchemeClassifier());
+    input.set_focus_type(OmniboxFocusType::DEFAULT);
+    return input;
+  }
+
+  // AutocompleteProviderListener:
+  void OnProviderUpdate(bool updated_matches) override {
+    if (autocomplete_->done())
+      base::RunLoop::QuitCurrentWhenIdleDeprecated();
+  }
+
+ protected:
+  // testing::Test
+  void SetUp() override { ASSERT_TRUE(SetUpImpl(true)); }
+
+  void TearDown() override {
+    autocomplete_ = nullptr;
+    client_.reset();
+    task_environment_.RunUntilIdle();
```

Comment:
what is this for?

Reply:
Unsure, it came over with everything else from the Chromium tests. Removed and it seems to still work :smile: